Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -136,16 +138,18 @@ class TopSitesPanel: UIViewController {
     }
 
     private func deleteHistoryTileForSite(site: Site, atIndexPath indexPath: NSIndexPath) {
-        func reloadThumbnails() {
-            self.profile.history.getTopSitesWithLimit(self.layout.thumbnailCount)
-                .uponQueue(dispatch_get_main_queue()) { result in
+        func reloadThumbnails() -> Success {
+            return self.profile.history.getTopSitesWithLimit(self.layout.thumbnailCount)
+                .bindQueue(dispatch_get_main_queue()) { result in
                     self.deleteOrUpdateSites(result, indexPath: indexPath)
```

Comment:
Hmm interesting. We can always do two separate performBatchUpdate calls - one for the deletion and another for the replacement. The only requirement is that the data source be in sync before/after each one. I'm going to give that a try. 

Also for https://bugzilla.mozilla.org/show_bug.cgi?id=1226201 I'd like to refactor the panel code to make it more testable so I can add a unit test for that bug so I'll probably do that as part of this work.